From 512c8d3c35bd5255b2f611b8c52b68bd2f5d5b2c Mon Sep 17 00:00:00 2001
From: Yuantian Tang <andy.tang@nxp.com>
Date: Thu, 24 Apr 2025 02:34:10 -0700
Subject: [PATCH] fix issue failed to boot kernel itb image

imx8mm, imx8mpevk, imx8mpfrdm, imx93evk, imx93frdm,
imx91evk and imx91frdm will be affected.

Signed-off-by: Andy Tang <andy.tang@nxp.com>
---
 configs/imx8mm_evk_defconfig             | 1 +
 configs/imx8mp_evk_defconfig             | 2 +-
 configs/imx8mp_frdm_defconfig            | 2 +-
 configs/imx91_11x11_evk_defconfig        | 2 ++
 configs/imx91_11x11_frdm_defconfig       | 2 ++
 configs/imx91_11x11_frdm_storm_defconfig | 2 ++
 configs/imx93_11x11_evk_defconfig        | 2 ++
 configs/imx93_11x11_frdm_defconfig       | 2 ++
 8 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/configs/imx8mm_evk_defconfig b/configs/imx8mm_evk_defconfig
index 703e2f49514..05f58ecd343 100644
--- a/configs/imx8mm_evk_defconfig
+++ b/configs/imx8mm_evk_defconfig
@@ -32,6 +32,7 @@ CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_BOARD_LATE_INIT=y
 CONFIG_SPL_HAS_BSS_LINKER_SECTION=y
 CONFIG_SPL_BSS_START_ADDR=0x910000
diff --git a/configs/imx8mp_evk_defconfig b/configs/imx8mp_evk_defconfig
index 7112fdf5859..70b23904a76 100644
--- a/configs/imx8mp_evk_defconfig
+++ b/configs/imx8mp_evk_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_frdm_defconfig b/configs/imx8mp_frdm_defconfig
index 0884b08ab2c..597ce28d00b 100644
--- a/configs/imx8mp_frdm_defconfig
+++ b/configs/imx8mp_frdm_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx91_11x11_evk_defconfig b/configs/imx91_11x11_evk_defconfig
index c017ccc9ace..47464f6ec7f 100644
--- a/configs/imx91_11x11_evk_defconfig
+++ b/configs/imx91_11x11_evk_defconfig
@@ -30,6 +30,8 @@ CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
diff --git a/configs/imx91_11x11_frdm_defconfig b/configs/imx91_11x11_frdm_defconfig
index bfa3e7cab4b..588dabe4554 100644
--- a/configs/imx91_11x11_frdm_defconfig
+++ b/configs/imx91_11x11_frdm_defconfig
@@ -30,6 +30,8 @@ CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
diff --git a/configs/imx91_11x11_frdm_storm_defconfig b/configs/imx91_11x11_frdm_storm_defconfig
index 81408bdf91a..84cb1fb2958 100644
--- a/configs/imx91_11x11_frdm_storm_defconfig
+++ b/configs/imx91_11x11_frdm_storm_defconfig
@@ -30,6 +30,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_defconfig b/configs/imx93_11x11_evk_defconfig
index 93b5c5a6575..287707466c8 100644
--- a/configs/imx93_11x11_evk_defconfig
+++ b/configs/imx93_11x11_evk_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_frdm_defconfig b/configs/imx93_11x11_frdm_defconfig
index bdfe6d905b9..cac592c8e10 100644
--- a/configs/imx93_11x11_frdm_defconfig
+++ b/configs/imx93_11x11_frdm_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
-- 
2.34.1

